catalyst/dl/callbacks/logging.py

```diff
@@ -183,6 +186,92 @@ def on_stage_end(self, state: RunnerState):
             logger.close()
 
 
+class TelegramLogger(Callback):
+    """
+    Logger callback, translates state.metrics to telegram channel
+    """
+
+    def __init__(
+        self,
+        token: str,
+        chat_id: str,
+        log_on_stage_start: bool = True,
+        log_on_loader_start: bool = True,
+        log_on_loader_end: bool = True,
+        log_on_stage_end: bool = True
+    ):
+        """
+
+        :param token: Bot's token. You can create bot by following this official guide: https://core.telegram.org/bots
+        :param chat_id: Chat unique identifier. You can get chat_id by following this guide:
+            1. Add your bot to a group.
+            2. Send a dummy message like this: /test @your_bot_name_here
+            3. Open this page: https://api.telegram.org/bot<token>/getUpdates, where <token> is yours bot token.
+                example: https://api.telegram.org/bot1234566789/getUpdates, where token = 1234566789.
+            4. You should look for this pattern ..."chat":{"id":<chat_id>..., where <chat_id> - id of your group.
+        :param log_on_stage_start: Send notification on stage start.
+        :param log_on_loader_start: Send notification on loader start.
+        :param log_on_loader_end: Send notification on loader end.
+        :param log_on_stage_end: Send notification on stage end.
+        """
+        super().__init__(CallbackOrder.Logger)
+        self._token = token
+        self._chat_id = chat_id
+        self._base_url = f"https://api.telegram.org/bot{self._token}/sendMessage"
+
+        self._log_on_stage_start = log_on_stage_start
+        self._log_on_loader_start = log_on_loader_start
+        self._log_on_loader_end = log_on_loader_end
+        self._log_on_stage_end = log_on_stage_end
+
+        self._metrics_to_log: Optional[List[str]] = None
+
+    def _send_text(self, text: str):
+        url = f"{self._base_url}?chat_id={self._chat_id}&disable_web_page_preview=1&text={quote_plus(text, safe='')}"
+
+        request = Request(url)
```

Comment:
It seems that pipeline can fall with error in case of problems with an internet connection. Can you check if `TelegramLogger` suffers from this and add something like `try-except` if required?

Reply:
done